src/Common/tests/System/Diagnostics/RemoteExecutorConsoleApp/RemoteExecutorConsoleApp.cs

```diff
@@ -69,10 +71,21 @@ static void Main(string[] args)
                 (instance as IDisposable)?.Dispose();
             }
 
-            // Use Exit rather than simply returning the exit code so that we forcibly shut down
-            // the process even if there are foreground threads created by the operation that would
-            // end up keeping the process alive potentially indefinitely.
-            Environment.Exit(exitCode);
+            // Environment.Exit not supported on .Net Native - don't even call it to avoid the nuisance exception.
+            if (!RuntimeInformation.FrameworkDescription.StartsWith(".NET Native", StringComparison.OrdinalIgnoreCase))
+            {
+                // Use Exit rather than simply returning the exit code so that we forcibly shut down
+                // the process even if there are foreground threads created by the operation that would
+                // end up keeping the process alive potentially indefinitely.
+                try
+                {
+                    Environment.Exit(exitCode);
+                }
+                catch (PlatformNotSupportedException)
+                {
+                }
```

Comment:
Meaning, I don't want to rely only on that RuntimeInfo check. I don't know every platform that doesn't (or won't in the future) support Environment.Exit.

The RuntimeInfoCheck is only there to prevent the minor nuisance of an extra first-chance exception during debugging.

The PNSE is the catch-all that prevents the far bigger problem of not being able to run or investigate a test because of the dialog box spam.